### DISCUSSION TITLE
`MemoryStore` Session Expiry

### DISCUSSION THREAD

**Discussion by SitanHuang** [maxcountryman/tower-sessions#33] [Q&A] [answered]
Hi,

It's not clear in the documentation whether `MemoryStore` handles clean up of stale sessions. Section "Extractor pattern" does not clearly indicate a way to configure `SessionManagerLayer` to set expiration of the extracted `Session` object.

Would appreciate any guidance on this.

**Comment by maxcountryman** [comment, marked as answer]
> It's not clear in the documentation whether MemoryStore handles clean up of stale sessions.

It does not. It's not intended to be used outside of testing and demonstration use cases. That said, if you want an in-memory only store that is appropriate for other use cases, have a look at [`MokaStore`](https://docs.rs/tower-sessions/latest/tower_sessions/struct.MokaStore.html). Moka can be bounded to a total number of entries and technically does support expiration although that is not currently implemented in tower-sessions.

**Reply by SitanHuang** [reply]
Wow, thanks for replying so quickly, and thanks for pointing out Moka. You prevented me from spending hours to implement something that already exists!
